core/src/main/scala/com/pingcap/tispark/telemetry/Telemetry.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pingcap.tispark.telemetry
+
+import com.fasterxml.jackson.databind.{DeserializationFeature, ObjectMapper}
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.pingcap.tispark.utils.HttpClientUtil
+import org.slf4j.LoggerFactory
+
+/**
+ * Report telemetry by HTTP POST. The url should be constant
+ */
+class Telemetry {
+  private val logger = LoggerFactory.getLogger(getClass.getName)
+  private var url = "https://telemetry.pingcap.com/api/v1/tispark/report"
+
+  /**
+   * Send telemetry message.
+   *
+   * @param msg the msg sent to telemetry server
+   */
+  def report(msg: TeleMsg): Unit = {
+    val httpClient = new HttpClientUtil
```

Comment:
will httpclient be closed automatically? If not, we should close it in finally

Reply:
yes, it would. Only httpserver needs to close actively.